nipype/pipeline/engine/workflows.py

```diff
@@ -1043,3 +1044,109 @@ def _get_dot(self,
                                                   vname1.replace('.', '_')))
                     logger.debug('cross connection: %s', dotlist[-1])
         return ('\n' + prefix).join(dotlist)
+
+    def add(self, name, node_like):
+        if is_interface(node_like):
+            node = Node(node_like, name=name)
+        elif is_node(node_like):
+            node = node_like
+
+        self.add_nodes([node])
+
+
+class Map(Node):
+    pass
+
+
+class Join(Node):
+    pass
+
+
+class MapState(object):
+    pass
+
+class NewNode(EngineBase):
+    def __init__(self, inputs={}, map_on=None, join_by=None,
+                 *args, **kwargs):
+        self._mappers = {}
+        self._joiners = {}
+
+    def map(self, field, values=None):
```

Comment:
I understand that `field` is a "mapper" itself (either list, tuple or string), is that right? Not sure about `values`, do you want to be a dictionary? From the Satra example I understand that you can provide values of some of the fields from the mapper (i.e. `field` in your syntax).

Reply:
I'm not actually positive of what I was doing here. Probably should have responded more promptly. :-) In any event, I clearly didn't finish this function (see L1076).

I believe what I was suggesting is that you could do:

```Python
node.map('a', [1, 2, 3, 4])

# or, where node.inputs.a takes the output of another node
node.map('a')
```